### PR TITLE
Extend find --show-pack-id to work with --tree

### LINKED ISSUE #2229: Make "restic find --show-pack-id" work with --tree
(I have a patch for this, just creating an issue so I can add a changelog.)

Useful to locate the pack that contains a tree.  This makes it easy to consistently create a repository missing a tree object, for testing #2224.

### PR DESCRIPTION
**Pull request by cdhowie**
What is the purpose of this change? What does it change?
--------------------------------------------------------
Extends `find --show-pack-id` to work with `--tree`, this resolves #2229.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------
This was not discussed, but was helpful to me when constructing a broken repository for #2224.

Checklist
---------
- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [ ] I have added tests for all changes in this PR
- [ ] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

Comments on above:

* No tests because I couldn't find any tests for `--show-pack-id` at all, so I'm not sure if it's necessary.
* I updated the built-in help documentation for the option, but didn't update the manual because the manual doesn't mention `--show-pack-id`.

### CODE REVIEW

**Review by fd0 — APPROVED** [review]
The code looks good and is approved! Since it's such an obscure feature, I wonder: is it better to not have a changelog entry for this? After all, I doubt this change is relevant for users...

**Comment by cdhowie** [comment]
I'll leave that up to you since you're the maintainer. It was on the checklist so I did it. :)

**Comment by fd0** [comment]
Ok, let's leave it out for now :)

**Pull request merged**
